pkg/k8s/watchers/watcher.go

```diff
@@ -252,8 +254,8 @@ type K8sWatcher struct {
 	podStoreSet  chan struct{}
 	podStoreOnce sync.Once
 
-	ciliumNodeStoreMU lock.RWMutex
-	ciliumNodeStore   cache.Store
+	ciliumNodeStore   resource.Store[*cilium_v2.CiliumNode]
```

Comment:
(optional)💡 based on the usage it seems we could use an atomic pointer here, one less mutex lock to think about.

Reply:
That's right, I've updated the code to use an `atomic.Pointer` :+1: